Index: nanovg/CMakeLists.txt
===================================================================
--- nanovg.orig/CMakeLists.txt
+++ nanovg/CMakeLists.txt
@@ -5,7 +5,7 @@ SET(VERSION 0.1)
 INCLUDE(GNUInstallDirs)
 
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/src)
-ADD_COMPILE_OPTIONS(-Wall -Werror -Wno-unused-but-set-variable -Wno-unused-variable)
+ADD_COMPILE_OPTIONS(-Wall -Werror)
 ADD_LINK_OPTIONS(LINKER:--as-needed LINKER:--no-undefined LINKER:--no-allow-shlib-undefined)
 
 SET(NANOVG_DEFINES "")
@@ -93,47 +93,3 @@ INSTALL(DIRECTORY ${CMAKE_SOURCE_DIR}/sr
 INSTALL(FILES ${CMAKE_BINARY_DIR}/nanovg.pc
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 
-# Build Examples
-
-ADD_EXECUTABLE(example_gl2
-	example/example_gl2.c example/demo.c example/perf.c
-	$<TARGET_OBJECTS:nanovg> $<TARGET_OBJECTS:nanovg_gl2>)
-TARGET_LINK_LIBRARIES(example_gl2 PRIVATE nanovg nanovg_gl2 GLEW EGL GL glfw m)
-TARGET_COMPILE_DEFINITIONS(example_gl2 PRIVATE ${NANOVG_GL_DEFINES} NANOVG_GL2)
-
-ADD_EXECUTABLE(example_gl3
-	example/example_gl3.c example/demo.c example/perf.c
-	$<TARGET_OBJECTS:nanovg> $<TARGET_OBJECTS:nanovg_gl3>)
-TARGET_LINK_LIBRARIES(example_gl3 PRIVATE nanovg nanovg_gl3 GLEW EGL GL glfw m)
-TARGET_COMPILE_DEFINITIONS(example_gl3 PRIVATE ${NANOVG_GL_DEFINES} NANOVG_GL3)
-
-ADD_EXECUTABLE(example_fbo example/example_fbo.c example/demo.c example/perf.c $<TARGET_OBJECTS:nanovg> $<TARGET_OBJECTS:nanovg_gl3>)
-TARGET_LINK_LIBRARIES(example_fbo PRIVATE nanovg_gl3 GLEW EGL GL glfw m)
-TARGET_COMPILE_DEFINITIONS(example_fbo PRIVATE ${NANOVG_GL_DEFINES} NANOVG_GL3)
-
-ADD_EXECUTABLE(example_gles2
-	example/example_gles2.c example/demo.c example/perf.c
-	$<TARGET_OBJECTS:nanovg> $<TARGET_OBJECTS:nanovg_gles2>)
-TARGET_LINK_LIBRARIES(example_gles2 PRIVATE GLEW EGL GL glfw m)
-TARGET_COMPILE_DEFINITIONS(example_gles2 PRIVATE NANOVG_GLES2)
-
-ADD_EXECUTABLE(example_gles3
-	example/example_gles3.c example/demo.c example/perf.c
-	$<TARGET_OBJECTS:nanovg> $<TARGET_OBJECTS:nanovg_gles3>)
-TARGET_LINK_LIBRARIES(example_gles3 PRIVATE GLEW EGL GL glfw m)
-TARGET_COMPILE_DEFINITIONS(example_gles3 PRIVATE NANOVG_GLES3)
-
-ADD_EXECUTABLE(example_svg1 example/example_svg1.c $<TARGET_OBJECTS:nanovg>)
-TARGET_LINK_LIBRARIES(example_svg1 PRIVATE GLEW EGL GL glfw m)
-
-ADD_EXECUTABLE(example_svg2 example/example_svg2.c $<TARGET_OBJECTS:nanovg>)
-TARGET_LINK_LIBRARIES(example_svg2 PRIVATE GLEW EGL GL glfw m)
-
-ADD_EXECUTABLE(example_oui
-	example/example_oui.c
-	$<TARGET_OBJECTS:nanovg> $<TARGET_OBJECTS:nanovg_gl3> $<TARGET_OBJECTS:nanovg_oui>)
-TARGET_LINK_LIBRARIES(example_oui PRIVATE GLEW EGL GL glfw m)
-TARGET_COMPILE_DEFINITIONS(example_oui PRIVATE DATADIR="../data")
-
-ADD_EXECUTABLE(example_gl_wrapper example/example_gl_wrapper example/demo.c $<TARGET_OBJECTS:nanovg>)
-TARGET_LINK_LIBRARIES(example_gl_wrapper PRIVATE nanovg nanovg_gl2 nanovg_gl3 nanovg_gles2 nanovg_gles3 GLEW EGL GL glfw m)
