Index: nanovg/CMakeLists.txt
===================================================================
--- nanovg.orig/CMakeLists.txt
+++ nanovg/CMakeLists.txt
@@ -142,6 +142,14 @@ INSTALL(DIRECTORY ${CMAKE_SOURCE_DIR}/sr
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/nanovg
         FILES_MATCHING PATTERN "nano*.h*")
 
+INSTALL(DIRECTORY ${CMAKE_SOURCE_DIR}/src/
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/nanovg
+        FILES_MATCHING PATTERN "oui*.h*")
+
+INSTALL(DIRECTORY ${CMAKE_SOURCE_DIR}/src/
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/nanovg
+        FILES_MATCHING PATTERN "blendish*.h*")
+
 # Install pkg-config files
 INSTALL(FILES ${CMAKE_BINARY_DIR}/nanovg.pc
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
