Index: harfbuzz/src/hb-ft.h
===================================================================
--- harfbuzz.orig/src/hb-ft.h
+++ harfbuzz/src/hb-ft.h
@@ -98,7 +98,7 @@ hb_ft_face_create_referenced (FT_Face ft
 
 /* See notes on hb_ft_face_create().  Same issues re lifecycle-management
  * apply here.  Use hb_ft_font_create_referenced() if you can. */
-HB_EXTERN hb_font_t *
+hb_font_t *
 hb_ft_font_create (FT_Face           ft_face,
 		   hb_destroy_func_t destroy);
 
