Index: stb/stb_image.h
===================================================================
--- stb.orig/stb_image.h
+++ stb/stb_image.h
@@ -4120,7 +4120,7 @@ static int stbi__zhuffman_decode_slowpat
    if (s >= 16) return -1; // invalid code!
    // code size is s, so:
    b = (k >> (16-s)) - z->firstcode[s] + z->firstsymbol[s];
-   if (b >= sizeof (z->size)) return -1; // some data was corrupt somewhere!
+   if (b >= (int) sizeof (z->size)) return -1; // some data was corrupt somewhere!
    if (z->size[b] != s) return -1;  // was originally an assert, but report failure instead.
    a->code_buffer >>= s;
    a->num_bits -= s;
