Index: nanogui-21e5cbc880b2e26b28b2a35085a9e6706da1e2a8/include/nanogui/layout.h
===================================================================
--- nanogui-21e5cbc880b2e26b28b2a35085a9e6706da1e2a8.orig/include/nanogui/layout.h
+++ nanogui-21e5cbc880b2e26b28b2a35085a9e6706da1e2a8/include/nanogui/layout.h
@@ -410,8 +410,8 @@ public:
 
         /// Allows for printing out Anchor position, size, and alignment.
         operator std::string() const {
-            char buf[50];
-            std::snprintf(buf, 50, "Format[pos=(%i, %i), size=(%i, %i), align=(%i, %i)]",
+            char buf[60];
+            std::snprintf(buf, 60, "Format[pos=(%i, %i), size=(%i, %i), align=(%i, %i)]",
                 pos[0], pos[1], size[0], size[1], (int) align[0], (int) align[1]);
             return buf;
         }
