Index: nanogui-0.0.20200404+0miry1/CMakeLists.txt
===================================================================
--- nanogui-0.0.20200404+0miry1.orig/CMakeLists.txt
+++ nanogui-0.0.20200404+0miry1/CMakeLists.txt
@@ -544,12 +544,17 @@ else()
 endif()
 
 if (NANOGUI_INSTALL)
+  configure_file(pkgconfig/nanogui.pc.in nanogui.pc @ONLY)
+
   install(TARGETS nanogui
           LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
           ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
   install(DIRECTORY include/nanogui DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
           FILES_MATCHING PATTERN "*.h")
+
+  install(FILES ${CMAKE_BINARY_DIR}/nanogui.pc
+          DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 endif()
 
 if (NANOGUI_BUILD_SHARED)
Index: nanogui-0.0.20200404+0miry1/pkgconfig/nanogui.pc.in
===================================================================
--- /dev/null
+++ nanogui-0.0.20200404+0miry1/pkgconfig/nanogui.pc.in
@@ -0,0 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@/nanogui
+
+Name: nanogui
+Description: Minimalistic cross-platform widget library
+Version: @VERSION@
+Requires: nanovg glew egl
+Libs: -L${libdir} -lnanogui -lGLU -lGL -lm
+Cflags: -I${includedir}
