Index: nanogui/src/example1.cpp
===================================================================
--- nanogui.orig/src/example1.cpp
+++ nanogui/src/example1.cpp
@@ -46,7 +46,7 @@
 #elif defined(__GNUC__)
 #   pragma GCC diagnostic ignored "-Wunused-function"
 #endif
-#include <stb_image.h>
+#include <stb/stb_image.h>
 
 using namespace nanogui;
 
Index: nanogui/src/texture.cpp
===================================================================
--- nanogui.orig/src/texture.cpp
+++ nanogui/src/texture.cpp
@@ -1,5 +1,5 @@
 #include <nanogui/texture.h>
-#include <stb_image.h>
+#include <stb/stb_image.h>
 #include <memory>
 
 NAMESPACE_BEGIN(nanogui)
